File: src/main/java/entities/gestionStore/facture.java
package entities.gestionStore;

import entities.gestionuser.User;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class facture {

    private int idFacture;
    private Date dateVente;
    private float prixtatalPaye;
    private String methodeDePaiement;
    private int id;
    private User leClient ;
    public List<detailfacture> ListeDetails = new ArrayList<>();


    public User getLeClient() {
        return leClient;
    }

    public void setLeClient(User leClient) {
        this.leClient = leClient;
    }

    public int getIdFacture() {
        return idFacture;
    }

    public void setIdFacture(int idFacture) {
        this.idFacture = idFacture;
    }

    public Date getDateVente() {
        return dateVente;
    }

    public void setDateVente(Date dateVente) {
        this.dateVente = dateVente;
    }

    public float getPrixtotalPaye() {
        return prixtatalPaye;
    }

    public void setPrixtotalPaye(float prixtotalPaye) {
        this.prixtatalPaye = prixtotalPaye;
    }

    public String getMethodeDePaiement() {
        return methodeDePaiement;
    }

    public void setMethodeDePaiement(String methodeDePaiement) {
        this.methodeDePaiement = methodeDePaiement;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public facture() {
    }

    public facture(int idFacture, Date dateVente, float prixtotalPaye, String methodeDePaiement, int id) {
        this.idFacture = idFacture;
        this.dateVente = dateVente;
        this.prixtatalPaye = prixtotalPaye;
        this.methodeDePaiement = methodeDePaiement;
        this.id = id;
    }

    public facture(int idFacture , String methodeDePaiement , int id)
    {
        this.idFacture = idFacture;
        this.methodeDePaiement = methodeDePaiement;
        this.id = id;
    }
    public facture(Date dateVente, float prixtotalPaye, String methodeDePaiement, int id) {
        this.dateVente = dateVente;
        this.prixtatalPaye = prixtotalPaye;
        this.methodeDePaiement = methodeDePaiement;
        this.id = id;
    }

    public facture(float prixtotalPaye, String methodeDePaiement, int id) {
        this.prixtatalPaye = prixtotalPaye;
        this.methodeDePaiement = methodeDePaiement;
        this.id = id;
    }

    public facture( String methodeDePaiement, int id) {
        this.methodeDePaiement = methodeDePaiement;
        this.id = id;
    }

    public facture( int idFacture , String methodeDePaiement)
    {
        this.methodeDePaiement = methodeDePaiement;
        this.idFacture = idFacture;
    }

    @Override
    public String toString() {
        return "facture{" +
                "idFacture=" + idFacture +
                ", dateVente=" + dateVente +
                ", prixtotalPaye=" + prixtatalPaye +
                ", methodeDePaiement='" + methodeDePaiement + '\'' +
                ", id=" + id +
                '}';
    }


    public float calculerPrixTotalFacture()
    {
        float prixTotal = 0.0f;
        for (detailfacture df : ListeDetails)
        {
            prixTotal += df.getPrixtotalArticle();
        }
        return prixTotal;
    }


}

File: webapp/src/GymPlus/src/Entity/Facture.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\FactureRepository;

#[ORM\Entity(repositoryClass: FactureRepository::class)]
class Facture
{
    // #[ORM\Id]
    // #[ORM\GeneratedValue]
    // #[ORM\Column]
    #[ORM\Id]
    #[ORM\Column(name: 'idFacture', type: 'integer')]
    #[ORM\GeneratedValue(strategy: 'AUTO')]
    private ?int $idfacture;

    #[ORM\Column]
    private ?\DateTime $datevente;

    #[ORM\Column]
    private ?float $prixtatalpaye;

    #[ORM\Column]
    private ?string $methodedepaiement;

  
//    #[ORM\ManyToOne(targetEntity: User::class)]
//    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id')]
   #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'factures')]
   #[ORM\JoinColumn(name: 'id', referencedColumnName: 'id', nullable: false)]

    private $user;//id;

    public function getIdfacture(): ?int
    {
        return $this->idfacture;
    }

    public function getDatevente(): ?\DateTime
    {
        return $this->datevente;
    }

    public function setDatevente(\DateTime $datevente): static
    {
        $this->datevente = $datevente;

        return $this;
    }

    public function getPrixtatalpaye(): ?float
    {
        return $this->prixtatalpaye;
    }

    public function setPrixtatalpaye(?float $prixtatalpaye): static
    {
        $this->prixtatalpaye = $prixtatalpaye;

        return $this;
    }

    public function getMethodedepaiement(): ?string
    {
        return $this->methodedepaiement;
    }

    public function setMethodedepaiement(?string $methodedepaiement): static
    {
        $this->methodedepaiement = $methodedepaiement;

        return $this;
    }

    public function getId(): ?User
    {
        return $this->user;//id;
    }

    public function setId(?User $user): static
    {
        $this->user = $user;

        return $this;
    }

    public function __construct()
    {
        $this->datevente = new \DateTime();

        //$this->ListeDetails = new ArrayCollection();

        
    }

    public function __toString(): string
    {
        return $this->getIdfacture();
    }

}
